core/src/SYCL/Kokkos_SYCL_Instance.hpp

```diff
@@ -239,13 +239,57 @@ class SYCLInternal {
   }
 };
 
+// FIXME_SYCL the limit is 2048 bytes for all arguments handed to a kernel,
+// assume for now that the rest doesn't need more than 248 bytes.
+#if defined(SYCL_DEVICE_COPYABLE) && defined(KOKKOS_ARCH_INTEL_GPU)
 template <typename Functor, typename Storage,
-          bool is_memcpyable = std::is_trivially_copyable_v<Functor>>
+          bool ManualCopy = (sizeof(Functor) >= 1800)>
```

Comment:
How did you pick `1800`?

Reply:
Admittedly, it's pretty random. So far we are only copying the functor and not the whole kernel to device-accessible space and so I need a little more space for the remaining kernel arguments to fit. Around 250 bytes feels pretty safe (even though parallel_for with `RangePolicy` only needs around 20 bytes) but I'm open to other suggestions.
In a follow-up pull request, I would try to consider the whole kernel when copying to device-accessible space, and then we should be able to use the full 2048 bytes here (maybe minus whatever space we need for `sycl::nd_item`).